src/main/java/frc/robot/systems/DriveFSMSystem.java

```diff
@@ -0,0 +1,337 @@
+package frc.robot.systems;
+
+// WPILib Imports
+
+
+// Third party Hardware Imports
+import com.revrobotics.CANSparkMax;
+import com.kauailabs.navx.frc.AHRS;
+
+// Robot Imports
+import frc.robot.TeleopInput;
+import frc.robot.HardwareMap;
+
+public class DriveFSMSystem {
+	/* ======================== Constants ======================== */
+	public static final double WHEEL_DIAMETER_INCHES = 7.65;
+	public static final double KP_MOVE_STRAIGHT = 0.1;
+	public static final double ERR_THRESHOLD_STRAIGHT_IN = 0.1;
+	private static final double TELEOP_ANGLE_POWER_RATIO = 90.0;
+	private static final double TURN_ERROR_POWER_RATIO = 360;
+	private static final double MIN_TURN_POWER = 0.1;
+	private static final double TURN_ERROR_THRESHOLD_DEGREE = 1.0;
+	private static final double COUNTS_PER_MOTOR_REVOLUTION = 42;
+	private static final double GEAR_RATIO = 26.0 * 4.67 / 12.0;
+	private static final double DRIVE_TICKS_PER_INCH
+		= COUNTS_PER_MOTOR_REVOLUTION * GEAR_RATIO / (Math.PI * WHEEL_DIAMETER_INCHES);
+	private static final double ODOMETRY_MIN_THETA = 1.0;
+
+	// FSM state definitions
+	public enum FSMState {
+		START_STATE,
+		FORWARD_STATE_10_IN,
+		TURN_STATE,
+		TELEOP_STATE
+	}
+
+	private static final float MOTOR_RUN_POWER = 0.1f;
+
+	/* ======================== Private variables ======================== */
+	private FSMState currentState;
+	private boolean finishedMovingStraight;
+	private boolean finishedTurning;
+	private double forwardStateInitialEncoderPos = -1;
+	private double rawGyroAngle = 0;
+	private double robotXPosLine = 0;
+	private double robotYPosLine = 0;
+	private double prevEncoderPosLine = 0;
+	private double prevEncoderPosArc = 0;
+	private double robotXPosArc = 0;
+	private double robotYPosArc = 0;
+	private double prevGyroAngle = 0;
+
+	// Hardware devices should be owned by one and only one system. They must
+	// be private to their owner system and may not be used elsewhere.
+
+	private CANSparkMax frontRightMotor;
+	private CANSparkMax backRightMotor;
+	private CANSparkMax frontLeftMotor;
+	private CANSparkMax backLeftMotor;
+
+	private AHRS gyro;
```

Comment:
It doesn't look like gyro gets instantiated anywhere. This is causing null pointer exceptions